src/data.rs

```diff
@@ -0,0 +1,320 @@
+/*
+    Copyright © 2023, ParallelChain Lab 
+    Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+//! Defines the data utilization of [hotstuff_rs::types::Data] (i.e. HotstuffData, a vector of vector of bytes) to serve the application. 
+//! The module also provides a struct called BlockData as a data abstraction of [hotstuff_rs::types::Block].data.
+//! 
+//! HotstuffData is a vector of vector of bytes. Each slot contains data in bytes (`Vec<u8>`). Block data is allocated into slots as below:
+//! 
+//! \[ BlockHeader Field 1, BlockHeader Field 2, ... , BlockHeader Field N, Tx 1, Tx 2, ... , Tx M, Receipt 1, Receipt 2, ... , Receipt M \]
+//! 
+//! where N is the number of fields in BlockHeader, M is the number of transactions (as well as receipts).
+
+use std::convert::{TryFrom, TryInto};
+
+use borsh::{BorshSerialize, BorshDeserialize};
+use hotstuff_rs::types::Data;
+
+use crate::{cryptography::{PublicAddress, Sha256Hash, BloomFilter, sha256}, serialization::{Serializable, Deserializable}, blockchain::{TransactionV2, ReceiptV2}};
+
+/// Indexes definitions for [hotstuff_rs::types::Block] interoperability.
+pub enum DatumIndexV2 {
```

Comment:
I was struggling because the original DatumIndex (V1) is declared and being used in another software component.
Moving the original code here seems unnecessary. (code looks more consistent at a cost of larger code size of a shared library used by other components)

Please advise.

Reply:
Is it declared in Fullnode? I wonder if I’m missing something.

Reply:
DatumIndexV1 is now defined in this pull request. Because @lyulka mentioned there could be application usage of this type.